### PR TITLE
Redirect stdout to output file for DOCKER_BUILDKIT builds

### PR DESCRIPTION
**Pull request by sonayak**
When building images using build kit, docker is redirecting the output to stderr instead of stdout. So no data is written to the output file which results in a warning message. As a fix, I am populating output logs from stderr if DOCKER_BUILDKIT variable is set to 1. 
This will fix issue #12470

### CODE REVIEW

File: Tasks/Common/docker-common-v2/dockercommandutils.ts [hunk 1]
```diff
@@ -38,7 +38,8 @@ export function build(connection: ContainerConnection, dockerFile: string, comma
 
     // setup variable to store the command output
     let output = "";
-    command.on("stdout", data => {
+    //In case of BuildKit build, docker tool is sending the logs to stderr.
+    command.on(isBuildKitBuild()?"stderr":"stdout", data => {
```
Comment (ds-ms) [inline Tasks/Common/docker-common-v2/dockercommandutils.ts:42]: Indent

File: Tasks/Common/docker-common-v2/dockercommandutils.ts [hunk 2]
```diff
@@ -377,4 +378,9 @@ function getDigest(imageId: string): string {
 
 function generateV2Name(input: string): string {
     return crypto.createHash("sha256").update(input).digest("hex");
+}
+
+function isBuildKitBuild(): boolean {
+    const isBuildKitBuildValue = tl.getVariable("DOCKER_BUILDKIT");
+    return isBuildKitBuildValue && Number(isBuildKitBuildValue)==1;
```
Comment (ds-ms) [inline Tasks/Common/docker-common-v2/dockercommandutils.ts:385]: Indent

File: Tasks/DockerV2/task.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "version": {
         "Major": 2,
         "Minor": 169,
```
Comment (ds-ms) [inline Tasks/DockerV2/task.json:16]: Set this Minor `170`

File: Tasks/Common/docker-common-v2/dockercommandutils.ts (hunk 1 above)
Comment (sonayak) [reply Tasks/Common/docker-common-v2/dockercommandutils.ts]: Fixed

**Pull request merged**